grails-app/services/org/broadinstitute/orsp/UserService.groovy

```diff
@@ -151,4 +152,64 @@ class UserService {
         }
     }
 
+    /**
+     * Edits a user's roles
+     *
+     * @param userId        The user's id
+     * @param rolesToAssign Roles String array to be assigned
+     */
+    void editUserRoles (Integer userId, ArrayList<String> rolesToAssign) {
+        if (userId != null) {
+            User user = User.findById(userId)
+            def validatedRoles = validateRoles(rolesToAssign)
+            if (user != null && validatedRoles.hasErrors) {
+                queryService.deleteOrspUserRoles(userId)
+                if (!rolesToAssign.isEmpty()) {
+                    queryService.updateOrspUserRoles(user, rolesToAssign)
+                }
+            } else {
+                log.error("Error while trying to modify roles to userId: ${userId}. ${validatedRoles.errorMessage}")
+                throw IllegalArgumentException()
+            }
+        } else {
+            log.error("Error while trying to modify roles to userId null.")
+            throw IllegalArgumentException("Cannot update roles of null userId")
+        }
+    }
+
+    /*
+     It would be nice to know what the error is here.
+      Can you pass that into the exception so it can bubble up?
+      Also, it would be nice to know why the failure happened.
+       Was the user null? What failed in the validation? etc.
+
+
+    */
```

Comment:
do we need this comment?

Reply:
Omg, sorry for that, i've must have copied that there for following up a suggestion. This is gone!